From 07c8300e75a338a62a1dd1fa1685e3fe271669a3 Mon Sep 17 00:00:00 2001
From: adi766 <adityakumarteli7846@gmail.com>
Date: Sat, 23 Sep 2017 23:27:29 +0000
Subject: [PATCH] Read MediaTek Serial number

---
 init/init.cpp | 35 +++++++++++++++++++++++++++++++++++
 1 file changed, 35 insertions(+)

diff --git a/init/init.cpp b/init/init.cpp
index ee5f03db3..0b3a45ecf 100755
--- a/init/init.cpp
+++ b/init/init.cpp
@@ -323,6 +323,37 @@ static int console_init_action(const std::vector<std::string>& args)
     return 0;
 }
 
+#ifdef MTK_HARDWARE
+static int USE_MTK_SERIAL()
+{
+    int serialdir;
+    char serial[32];
+    size_t m;
+
+    serialdir = open("/sys/sys_info/serial_number", O_RDWR);
+    if (serialdir < 0) {
+        NOTICE("fail to open:", "/sys/sys_info/serial_number");
+        return 0;
+    }
+    m = read(fd, serialno, sizeof(char)*32);
+
+    serial[s-1] = '\0';
+
+    close(serialdir);
+
+    if (m <= 0) {
+	    NOTICE("could not get mtk serial number from  sys file\n");
+	    return 0;
+	}
+
+    NOTICE("serial number=%s\n",serial);
+
+    property_set("ro.boot.serialno", serial);
+
+    return 1;
+}
+#endif
+
 static void import_kernel_nv(const std::string& key, const std::string& value, bool for_emulator) {
     if (key.empty()) return;
 
@@ -421,6 +452,10 @@ static void process_kernel_cmdline() {
     // as properties.
     import_kernel_cmdline(false, import_kernel_nv);
     if (qemu[0]) import_kernel_cmdline(true, import_kernel_nv);
+
+#ifdef MTK_HARDWARE
+    USE_MTK_SERIAL();
+#endif
 }
 
 static int property_enable_triggers_action(const std::vector<std::string>& args)
